test-suite/js/src/test/scala/org/scalajs/testsuite/jsinterop/AsyncTest.scala

```diff
@@ -136,4 +199,86 @@ object AsyncTest extends JasmineTest {
 
   }
 
+  describe("js.Promise.toFuture") {
+    it("basic case") {
+      PromiseMock.withMockedPromise { processQueue =>
+        implicit val ec = QueueExecutionContext.promises()
+
+        val p = new js.Promise[Int]({
+          (resolve: js.Function1[Int | js.Thenable[Int], _], reject: js.Function1[Any, _]) =>
+            resolve(42)
+        })
+
+        val f = p.toFuture
+        val fAssertType: Future[Int] = f
```

Comment:
Hahahaha, local type inference FTW :) Try doing this in a global inferencer without influencing the `p.toFuture` call.